src/TorchSharp/Distributions/GumbEL.cs

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) .NET Foundation and Contributors.  All Rights Reserved.  See LICENSE in the project root for license information.
+using System;
+using System.Collections.Generic;
+using static TorchSharp.torch;
+
+namespace TorchSharp
+{
+    using System.Linq;
+    using Modules;
+    using TorchSharp.torchvision;
+    using static torch.distributions;
+
+    namespace Modules
+    {
+        /// <summary>
+        /// Samples from a Gumbel Distribution.
+        /// </summary>
+        public class Gumbel : TransformedDistribution
+        {
+            internal Gumbel(Tensor loc, Tensor scale, Distribution base_distribution, torch.distributions.transforms.Transform[] transforms, torch.Generator generator = null) :
+                base(base_distribution, transforms, generator)
+            {
+                this.batch_shape = loc.size();
+                var locScale = torch.broadcast_tensors(loc, scale);
+                this.loc = locScale[0];
+                this.scale = locScale[1];
+            }
+
+            private Tensor loc;
+            private Tensor scale;
+
+            public override Tensor mean => loc + scale * euler_constant;
+
+            public override Tensor mode => loc;
+
+            public override Tensor variance => stddev.pow(2);
+
+            public override Tensor stddev => pioversqrtsix * scale;
+
+            /// <summary>
+            /// Returns entropy of distribution, batched over batch_shape.
+            /// </summary>
+            public override Tensor entropy() => scale.log() + (1 + euler_constant);
+
+            private double pioversqrtsix = Math.PI / Math.Sqrt(6);
```

Comment:
>pioversqrtsix [](http://example.com/codeflow?start=27&length=13)

this should be const?

Reply:
Yes, it should. Thanks!

Reply:
Take that back -- it's not initialized to a compile-time constant, so that will generate and error. 'readonly,' perhaps?

Reply:
I meant calculate it offline and use the value. 

```C#
private const pioversqrtsix  = 1.2825498301618641; //  Math.PI / Math.Sqrt(6);
```

Reply:
Of course. Thanks!